### PR TITLE
Refactor repository to follow standard repo pattern

### PR DESCRIPTION
**Pull request by bojanaarmacki**
- Add `FeatureARepository` interface and implementation
- `FeatureAViewModel` to reference `FeatureARepository`
- Add `FeatureALocalDataSource` and `FeatureARemoteDataSource` classes
- Add `model`, `repository` and `source` package to `data`
- Remove `service` package
- Move dependencies' version declarations from app-level `build.gradle` file to `libs.versions.toml`
- Start migration support for `Room` - specify `schemas` folder 
- Add `DispatchersModule` to inject `CoroutineDispatchers.IO` and `CoroutineDispatchers.Default`

### CODE REVIEW

File: app/src/main/java/com/blueprint/featureA/data/repository/DefaultFeatureALocalRepository.kt [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+) : FeatureALocalRepository {
+    override suspend fun insertFeatureA(featureAEntity: FeatureAEntity) {
+        featureADao.insertFeatureA(featureAEntity)
+    }
+
+    override suspend fun updateFeatureA(featureAEntity: FeatureAEntity) {
+        featureADao.updateFeatureA(featureAEntity)
+    }
+
+    override suspend fun getFeatureAById(id: Int): FeatureAEntity? {
```
[the remark this answers is not in the document]
Comment (bojanaarmacki) [reply app/src/main/java/com/blueprint/featureA/data/repository/DefaultFeatureALocalRepository.kt]: E da, ova klasa je obrisana u poslednjem komitu.
Slazem se, flow se savetuje kao best practice [source](https://developer.android.com/kotlin/coroutines/coroutines-best-practices?_gl=1*1041cqc*_up*MQ..&gclid=CjwKCAjwhNWZBhB_EiwAPzlhNtReVIBfrUFBUt6SqZz3YLezP9YEiGuBube4YSTrOF-0ovxzpNGNaRoCiYsQAvD_BwE&gclsrc=aw.ds#coroutines-data-layer)

"Classes in the data and business layers generally expose functions to perform one-shot calls or to be notified of data changes over time. _Classes in those layers should expose suspend functions for one-shot calls and Flow to notify about data changes._"

**Review by StefanPD — APPROVED** [review]

File: app/src/main/java/com/blueprint/featureA/data/repository/DefaultFeatureALocalRepository.kt (hunk 1 above)
[the remark this answers is not in the document]
Comment (StefanPD) [reply app/src/main/java/com/blueprint/featureA/data/repository/DefaultFeatureALocalRepository.kt]: Ne moraju biti suspend ali zavisi sta hocemo da postignemo.
Non-suspend ce blokirati taj thread dok suspend nece. Generalno ako ocekujemo da ce neki query biti long term operacija trebali bi koristiti suspend

[the remark this answers is not in the document]
Comment (ivannenadovic) [reply app/src/main/java/com/blueprint/featureA/data/repository/DefaultFeatureALocalRepository.kt]: Da. Recimo to moze da se desi kod query-ija koji rade vise stvari. Tipa podaci s mreze se čuvaju u room, pa onda mi čupamo iz Rooma. A to sve u okviru istog poziva iz VM. 
Tada ima smisla koristiti suspend.

**Review by ivannenadovic — APPROVED** [review]
LGTM

**Pull request merged**